whatlies/embeddingset.py

```diff
@@ -178,7 +200,24 @@ def __or__(self, other):
         (emb | buz).plot(kind="arrow")
         ```
         """
-        new_embeddings = {k: emb | other for k, emb in self.embeddings.items()}
+        if isinstance(other, Embedding):
+            new_embeddings = {k: emb | other for k, emb in self.embeddings.items()}
+        elif isinstance(other, EmbeddingSet):
+            # Apply Gram-Schmidt to project away from a hyperplane instead of an axis
+            # First, create orthogonal vectors that span the space to project away from
+            orth_away = [other[w] for w in other]
+            for i in range(len(orth_away)):
+                orth_away[i] = reduce(lambda a, b: b | a, orth_away[: i + 1])
```

Comment:
The import line for `reduce` is missing.

Reply:
Are you sure? The tests wouldn't pass if it were and locally at least it seems to be imported on line 2.

Reply:
Sorry! That's right. I thought we haven't used it already in this file.